sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcIOIT.java

```diff
@@ -0,0 +1,186 @@
+ * set exists in the database - if not, you can run JdbcTestDataSet and it will create the read
+ * table for you.
+ *
+ * <p>You can run just this test by doing the following:
+ * mvn test-compile compile failsafe:integration-test -D beamTestPipelineOptions='[
+ * "--postgresIp=1.2.3.4",
+ * "--postgresUsername=postgres",
+ * "--postgresDatabaseName=myfancydb",
+ * "--postgresPassword=yourpassword",
+ * "--postgresSsl=false"
+ * ]' -DskipITs=false -Dit.test=org.apache.beam.sdk.io.jdbc.JdbcIOIT -DfailIfNoTests=false
+ */
+@RunWith(JUnit4.class)
+public class JdbcIOIT {
+
+  public PGSimpleDataSource getDataSource() throws SQLException {
+    PipelineOptionsFactory.register(PostgresTestOptions.class);
+    PostgresTestOptions options = TestPipeline.testingPipelineOptions()
+        .as(PostgresTestOptions.class);
+    PGSimpleDataSource dataSource = new PGSimpleDataSource();
+
+    // Tests must receive parameters for connections from PipelineOptions
+    // Parameters should be generic to all tests that use a particular datasource, not
+    // the particular test.
+    dataSource.setDatabaseName(options.getPostgresDatabaseName());
+    dataSource.setServerName(options.getPostgresIp());
+    dataSource.setPortNumber(options.getPostgresPort());
+    dataSource.setUser(options.getPostgresUsername());
+    dataSource.setPassword(options.getPostgresPassword());
+    dataSource.setSsl(options.getPostgresSsl());
+
+    return dataSource;
+  }
+
+  // These static classes are used since if they were inlined using anonymous classes, they would cause the JdbcIOIT
+  // class as a whole to be serialized.
+
+  private static class TestRowMapper implements JdbcIO.RowMapper<KV<String, Integer>> {
+    @Override
+    public KV<String, Integer> mapRow(ResultSet resultSet) throws Exception {
+      KV<String, Integer> kv =
+          KV.of(resultSet.getString("name"), resultSet.getInt("id"));
+      return kv;
+    }
+  }
+
+  private static class ValidateCountFn implements SerializableFunction<Iterable<KV<String, Long>>, Void> {
+    @Override
+    public Void apply(Iterable<KV<String, Long>> input) {
+      for (KV<String, Long> element : input) {
+        assertEquals(element.getKey(), 100L, element.getValue().longValue());
+      }
+      return null;
+    }
+  };
+
+  private static class TestStatementSetter implements JdbcIO.PreparedStatementSetter<KV<Integer, String>> {
+    @Override
+    public void setParameters(KV<Integer, String> element, PreparedStatement statement)
+                    throws SQLException {
+      statement.setInt(1, element.getKey());
+      statement.setString(2, element.getValue());
+    }
+  }
+
+  /**
+   * Does a test read of a few rows from a postgres database.
+   *
+   * <p>Note that IT read tests must not do any data table manipulation (setup/clean up.)
+   * @throws SQLException
+   */
+  @Test
+  @Category(NeedsRunner.class)
```

Comment:
I'm not sure how I feel about the use of `NeedsRunner` here - shouldn't `ITs` be invoked as part of the test infrastructure and suppressed as part of the maven build regardless of their `Category`?